src/hb-ot-cmap-table.hh

```diff
@@ -851,182 +929,62 @@ struct cmap
 {
   static constexpr hb_tag_t tableTag = HB_OT_TAG_cmap;
 
-  struct subset_plan
+  template<typename Iterator,
+           hb_requires (hb_is_iterator (Iterator))>
+  void serialize (hb_serialize_context_t *c,
+                  Iterator it,
+                  const EncodingRecord *has_unicode_bmp,
+                  const EncodingRecord *has_unicode_ucs4,
+                  const EncodingRecord *has_ms_bmp,
+                  const EncodingRecord *has_ms_ucs4)
```

Comment:
I find this api confusing.  These names sound like should be boolean, but they are actual records, which can be nullptr I guess.